### ISSUE TITLE
DELETE method regression

### ISSUE THREAD

**Issue by mattyr** [ruby-grape/grape-swagger#591]
Looks like #583 introduced a regression, where if you have a 200 response code with a model (entity) on a delete method, you get:

```sh
NoMethodError - undefined method `[]=' for nil:NilClass:
   () Users/matt/.rvm/gems/ruby-2.2.6@gorecess/bundler/gems/grape-swagger-e1cd8ae702f6/lib/grape-swagger/endpoint.rb:211:in `block in response_object'
   () Users/matt/.rvm/gems/ruby-2.2.6@gorecess/bundler/gems/grape-swagger-e1cd8ae702f6/lib/grape-swagger/endpoint.rb:193:in `response_object'
   () Users/matt/.rvm/gems/ruby-2.2.6@gorecess/bundler/gems/grape-swagger-e1cd8ae702f6/lib/grape-swagger/endpoint.rb:117:in `method_object'
   () Users/matt/.rvm/gems/ruby-2.2.6@gorecess/bundler/gems/grape-swagger-e1cd8ae702f6/lib/grape-swagger/endpoint.rb:97:in `block in path_item'
   () Users/matt/.rvm/gems/ruby-2.2.6@gorecess/bundler/gems/grape-swagger-e1cd8ae702f6/lib/grape-swagger/endpoint.rb:91:in `path_item'
   () Users/matt/.rvm/gems/ruby-2.2.6@gorecess/bundler/gems/grape-swagger-e1cd8ae702f6/lib/grape-swagger/endpoint.rb:76:in `block in path_and_definition_objects'
   () Users/matt/.rvm/gems/ruby-2.2.6@gorecess/bundler/gems/grape-swagger-e1cd8ae702f6/lib/grape-swagger/endpoint.rb:74:in `path_and_definition_objects'
   () Users/matt/.rvm/gems/ruby-2.2.6@gorecess/bundler/gems/grape-swagger-e1cd8ae702f6/lib/grape-swagger/doc_methods.rb:54:in `block in setup'
   () Users/matt/.rvm/gems/ruby-2.2.6@gorecess/bundler/gems/grape-swagger-e1cd8ae702f6/lib/grape-swagger/doc_methods.rb:68:in `block in setup'
  grape (0.17.0) lib/grape/endpoint.rb:59:in `block (2 levels) in generate_api_method'
```

will submit a pr with reproduction/fix in a moment

**Comment by LeFnord** [comment]
@mattyr … please upgrade first grape, the default status code for delete was changed from 200 to 204 in version 0.19.1, so with this version it should be correct

**Comment by mattyr** [comment]
already using 0.19.1:

```sh
➜  grape-swagger git:(delete_method_regression) bundle show
Gems included by the bundle:
  * activesupport (5.0.1)
  * addressable (2.5.0)
  * ast (2.3.0)
  * axiom-types (0.1.1)
  * builder (3.2.3)
  * bundler (1.14.5)
  * byebug (9.0.6)
  * claide (1.0.1)
  * claide-plugins (0.9.2)
  * coderay (1.1.1)
  * coercible (1.0.0)
  * colored (1.2)
  * concurrent-ruby (1.0.5)
  * cork (0.2.0)
  * coveralls (0.8.19)
  * danger (4.0.5)
  * danger-changelog (0.2.1)
  * danger-plugin-api (1.0.0)
  * descendants_tracker (0.0.4)
  * diff-lcs (1.3)
  * docile (1.1.5)
  * equalizer (0.0.11)
  * faraday (0.11.0)
  * faraday-http-cache (1.3.1)
  * git (1.3.0)
  * grape (0.19.1)
  * grape-entity (0.6.1)
  * grape-swagger (0.26.1)
  * grape-swagger-entity (0.1.6)
  * hashie (3.5.5)
  * i18n (0.8.1)
  * ice_nine (0.11.2)
  * json (2.0.3)
  * kramdown (1.13.2)
  * method_source (0.8.2)
  * minitest (5.10.1)
  * multi_json (1.12.1)
  * multi_xml (0.6.0)
  * multipart-post (2.0.0)
  * mustermann (0.4.0)
  * mustermann-grape (0.4.0)
  * nap (1.1.0)
  * octokit (4.6.2)
  * open4 (1.3.4)
  * parser (2.4.0.0)
  * powerpack (0.1.1)
  * pry (0.10.4)
  * pry-byebug (3.4.2)
  * public_suffix (2.0.5)
  * rack (2.0.1)
  * rack-accept (0.4.5)
  * rack-cors (0.4.1)
  * rack-test (0.6.3)
  * rainbow (2.2.1)
  * rake (12.0.0)
  * rdoc (5.1.0)
  * rspec (3.5.0)
  * rspec-core (3.5.4)
  * rspec-expectations (3.5.0)
  * rspec-mocks (3.5.0)
  * rspec-support (3.5.0)
  * rubocop (0.47.1)
  * ruby-grape-danger (0.1.1)
  * ruby-progressbar (1.8.1)
  * sawyer (0.8.1)
  * simplecov (0.12.0)
  * simplecov-html (0.10.0)
  * slop (3.6.0)
  * term-ansicolor (1.4.0)
  * terminal-table (1.7.3)
  * thor (0.19.4)
  * thread_safe (0.3.6)
  * tins (1.13.2)
  * tool (0.2.3)
  * tzinfo (1.2.2)
  * unicode-display_width (1.1.3)
  * virtus (1.0.5)
```

**Comment by mattyr** [comment]
if you try out the associated pull request, #592 , removing the code changes and just running the spec, you'll see the errors (hopefully!).

**Comment by mattyr** [comment]
this should be fixed with merge of #597
